src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -86,6 +86,19 @@ internal static class ParenthesizedExpressionSyntaxExtensions
                 return false;
             }
 
+            // Check for cases like in https://github.com/dotnet/roslyn/issues/43934
+            if (expression.IsKind(SyntaxKind.GreaterThanExpression, SyntaxKind.LessThanExpression) &&
+                nodeParent.Parent is TupleExpressionSyntax tupleExpression &&
+                tupleExpression.Arguments.Any(a => a.Expression is ParenthesizedExpressionSyntax parenthesizedExpression
```

Comment:
you don't need to check for parenthesized expressions.  that seems superflous.  What do you get if you remov ethat?

Reply:
the point of hte 'Count' check from before was to only block in the case where tehre was at least one < and one > thus raising the possibility of a generic interpretation.

Reply:
`Any` is the same as `Count() >= 1`, isn't it?

Reply:
> What do you get if you remov ethat?

Test of the code of the original issue fails, because both expressions are `ParenthesizedExpressionSyntax` there

Reply:
> Test of the code of the original issue fails, because both expressions are ParenthesizedExpressionSyntax there

it's fine for eitehr to be removed right?  it's only a problem if both are removed.

Reply:
> Any is the same as Count() >= 1, isn't it?

D'oh!  Brainfart.  Thanks :)

Reply:
> it's only a problem if both are removed

Yes, but here is the logic: analizer reports diagnostic and then codefix fixes it without "looking" at other places of the diagnostics. I verified, that because of this, if I remove the check, "Fix all" will produce invalid code:
![devenv_nckbyURg2o](https://user-images.githubusercontent.com/70431552/163625141-bb9779e3-8cec-47ee-8f37-52404073fc2f.png)
So yes, the code is bit more messy than it was originally planned